### PR TITLE
7.article id/comments

### PR DESCRIPTION
**Pull request by K-Othman**

### CODE REVIEW

File: __tests__/news.test.js [hunk 1]
```diff
@@ -29,3 +30,167 @@ describe.only("1. GET /api/topics", () => {
...
+      });
+  });
+  test("400: response with bad request if article_id is not a number", () => {
+    return request(app)
+      .get(`/api/articles/dog/comments`)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
```
Comment (MRombi) [inline __tests__/news.test.js:322]: We are missing a few extra tests:

- status 201 and ignores extra properties i.e. 
const newComment = {
      username: "butter_bridge",
      body: "Oh, I've got compassion running out of my nose, pal!",
      banana: "fruit"
    };
    - status 404 if the id is a number but is not valid i.e. 9999
    - status 400 if missing required field(s), e.g. no username 
    - status 404 if username does not exists

**Review by MRombi — CHANGES_REQUESTED** [review]
Well done on this ticket, we have a few extra tests that need to be built alongside the code to make them pass. Please, action on the feedback and resubmit 🙂

File: __tests__/news.test.js [hunk 2]
```diff
@@ -29,3 +30,200 @@ describe.only("1. GET /api/topics", () => {
...
+      username: "butter_bridge",
+      body: "Oh, I've got compassion running out of my nose, pal!",
+    };
+    return request(app)
+      .post(`/api/articles/3/comments`)
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toEqual({
+          article_id: 3,
```
Comment (corkesw) [inline __tests__/news.test.js]: You should provide specific values here as we should know what they will be and should check for them. The only exception would be `created_at` as we won't know exactly what the time stamp is.

File: __tests__/news.test.js [hunk 3]
```diff
@@ -29,3 +30,200 @@ describe.only("1. GET /api/topics", () => {
...
+      body: "Oh, I've got compassion running out of my nose, pal!",
+    };
+    return request(app)
+      .get(`/api/articles/dog/comments`)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+  test("404: response with bad request if article_id is not a number", () => {
```
Comment (corkesw) [inline __tests__/news.test.js]: This test is a duplicate and of an earlier test

File: __tests__/news.test.js [hunk 4]
```diff
@@ -29,3 +30,200 @@ describe.only("1. GET /api/topics", () => {
...
+      body: "Oh, I've got compassion running out of my nose, pal!",
+    };
+    return request(app)
+      .get(`/api/articles/dog/comments`)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+});
```
Comment (corkesw) [inline __tests__/news.test.js:323]: We still need a 404 for a valid article_id but for an article that does not exist
eg `/api/articles/999/comments`

**Review by corkesw — COMMENTED** [review]
Still another test to add here

**Pull request merged**